libs/base/Data/Buffer.idr

```diff
@@ -45,153 +45,153 @@ newBuffer size
 %foreign "scheme:blodwen-buffer-setbyte"
          "RefC:setBufferByte"
          "node:lambda:(buf,offset,value)=>buf.writeUInt8(value, offset)"
-prim__setByte : Buffer -> Int -> Int -> PrimIO ()
+prim__setByte : Buffer -> (offset : Int) -> (val : Int) -> PrimIO ()
```

Comment:
Alternatively we could define:

```
Set : (Type -> Type) -> Type -> Type
Set io a = Buffer -> (offset : Int) -> a -> io ()
```

and reuse that type alias everywhere. This way no use site will mention
two types that could legitimately be understood as indices.

Reply:
OK, I tried it here https://github.com/stepancheg/Idris2/commit/2b352ec55ecba3c67c45156bcfb915487252d261 and actually it decreases code readability while I wanted to make code more readable.